### ISSUE TITLE
[BUG] `InteractiveBrowserCredentialBuilder` forces `clientId` to be `IdentityConstants.DEVELOPER_SINGLE_SIGN_ON_ID`

### ISSUE THREAD

**Issue by bdevreugd-vialis** [Azure/azure-sdk-for-java#20306]
**Describe the bug**
The builder [InteractiveBrowserCredentialBuilder](https://github.com/Azure/azure-sdk-for-java/blob/41bb40b96e1488ed68e73b8c63cd85296c239c10/sdk/identity/azure-identity/src/main/java/com/azure/identity/InteractiveBrowserCredentialBuilder.java) forces the `clientId` to be `IdentityConstants.DEVELOPER_SINGLE_SIGN_ON_ID`, destroying the ability to set your own clientId.

The builder methods for the `clientId` are located in the superclass ([AadCredentialBuilderBase](https://github.com/Azure/azure-sdk-for-java/blob/41bb40b96e1488ed68e73b8c63cd85296c239c10/sdk/identity/azure-identity/src/main/java/com/azure/identity/AadCredentialBuilderBase.java#L38-L41)), therefore if you set your own clientId it is being ignored and you will always get `IdentityConstants.DEVELOPER_SINGLE_SIGN_ON_ID`

**To Reproduce**
Authenticate using `InteractiveBrowserCredentialBuilder` with a custom `clientId`

***Code Snippet***
In the class: `InteractiveBrowserCredentialBuilder`, on [line 21](https://github.com/Azure/azure-sdk-for-java/blob/41bb40b96e1488ed68e73b8c63cd85296c239c10/sdk/identity/azure-identity/src/main/java/com/azure/identity/InteractiveBrowserCredentialBuilder.java#L21) the code sets the `clientId`:

```java
String clientId = IdentityConstants.DEVELOPER_SINGLE_SIGN_ON_ID;
```

**Expected behavior**
When I set my own `clientId`, I expect it to be honored.

**Comment by joshfree** [comment]
@g2vinay PTAL

**Comment by bdevreugd-vialis** [comment]
I can see it is fixed in code, thanks, however I have to use the 1.3.0 beta (because 1.2.5 does not prompt for account selection, nor has any option to do so) and the beta does not have the fix released yet.

**Comment by g2vinay** [comment]
@bdevreugd-vialis 
The account selection prompt feature/behavior is going to be released next week in version 1.3.0 of the azure-identity.
1.2.5 contains the fix where user-configured Client Id is not overridden by DEVELOPER_SINGLE_SIGN_ON_ID in Interactive Browser Credential. (the issue pointed by this ticket)
